lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -163,6 +200,41 @@ def compact_messages(batch)
 
         batch.reverse.uniq(&:key).reverse!
       end
+
+      # Turns Kafka payload into ActiveRecord Objects by mapping relevant fields
+      # Override this method to build object and associations with message payload
+      # @param messages [Array<Deimos::Message>] the array of deimos messages in batch mode
+      # @return [Array<ActiveRecord>] Array of ActiveRecord objects
+      def build_models(messages)
+        messages.map do |m|
+          attrs = if self.method(:record_attributes).parameters.size == 2
+                    record_attributes(m.payload, m.key)
+                  else
+                    record_attributes(m.payload)
+                  end
+
+          attrs&.merge(record_key(m.key))
+          @klass.new(attrs)
+        end
+      end
+
+      # Filters list of Active Records by applying active record validations. Optionally, validation_context can
+      # be set in ActiveRecordConsumer to override the default `nil` context.
+      # Tip: Add validates_associated in ActiveRecord model to validate associated_models
+      # Optionally inherit this method and apply more filters in the application code
+      # @param records Array<ActiveRecord> - List of active records which will be subjected to model validations
+      # @return valid Array<ActiveRecord> - Subset of records that passed the model validations
+      def filter_models(records)
+        valid, invalid = records.partition do |p|
+          p.valid?(@validation_context)
```

Comment:
I feel like adding a context here is actually not a common thing to do - usually things are either valid or they're not. I'd leave out the context here and override this in item-platform to add the context. It's a small enough method to be safe.

Reply:
Removed.